### PR TITLE
RavenDB-8060: Added limit to number of concurrent subscriptions conne…

### PR DESCRIPTION
**Pull request by maximburyak**
…ctions

### CODE REVIEW

File: src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+using Raven.Server.Config.Attributes;
+using Raven.Server.Config.Settings;
+
+namespace Raven.Server.Config.Categories
+{
+    public class SubscriptionConfiguration : ConfigurationCategory
+    {
+        [DefaultValue(10)]
+        [TimeUnit(TimeUnit.Minutes)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:13]: Why is this a time unit?

File: src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        [ConfigurationEntry("Subscriptions.ConcurrentConnections")]
+        [Description("Amount of concurrent subscription connections.")]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:15]: Per server? Per cluster? Per database?

File: src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using System.ComponentModel;
+using System.Text;
+using Raven.Server.Config.Attributes;
+using Raven.Server.Config.Settings;
+
+namespace Raven.Server.Config.Categories
+{
+    public class SubscriptionConfiguration : ConfigurationCategory
+    {
+        [DefaultValue(10)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:12]: How many subscriptions did you run on your machine before it became a big issue?

Comment (maximburyak) [reply src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:12]: I didn't try to "stress" it yet. But we did want to limit the number of concurrent connections, maybe 10 is a bit low for a default value.

Comment (ayende) [reply src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:12]: Yes, but we need to be aware what the actual limits are

Comment (maximburyak) [reply src/Raven.Server/Config/Categories/SubscriptionConfiguration.cs:12]: Apparently it's not that easy.. I've used the help of 5 of our teammates, and managed to run 1000 concurrent subscriptions, with a pretty good throughput (don't have the exact number, but it's more than a hundred empty documents per subscription. I think that I'll set the value to 1000 for now. Something else that bothers me is the client side cost: every 100 subscription connections consumes ~25% of the cpu and ~4 gigs of memory, but I think that we'll leave that to RavenDB-7283 (subscriptions benchmark issue)

**Pull request merged**